### PR TITLE
Removed h2 database and hibernate/jpa stuff

### PR DESCRIPTION
**Pull request by dmitryintel**

### CODE REVIEW

File: nsfc-plugin/pom.xml [hunk 1]
```diff
@@ -69,59 +69,29 @@
             <artifactId>nsfc-uber-openstack4j</artifactId>
             <version>2.0.1-SNAPSHOT</version>
         </dependency>
+
         <dependency>
             <groupId>org.glassfish.jersey.core</groupId>
```
Comment (arvindn05) [inline nsfc-plugin/pom.xml]: why is jersey still a dependency, even if its test only scope...there should not be any dependency.

File: nsfc-plugin/src/test/java/org/osc/controller/nsfc/OSGiIntegrationTest.java [hunk 2]
```diff
@@ -209,12 +177,9 @@ public SSLContext getSslContext() {
                     mavenBundle("org.apache.servicemix.bundles", "org.apache.servicemix.bundles.dom4j")
                             .versionAsInProject(),
                     mavenBundle("org.javassist", "javassist").versionAsInProject(),
-                    mavenBundle("org.jboss.logging", "jboss-logging").versionAsInProject(),
-                    mavenBundle("org.jboss", "jandex").versionAsInProject(),
+//                    mavenBundle("org.jboss.logging", "jboss-logging").versionAsInProject(),
```
Comment (arvindn05) [inline nsfc-plugin/src/test/java/org/osc/controller/nsfc/OSGiIntegrationTest.java]: remove commented

File: nsfc-plugin/src/main/java/org/osc/controller/nsfc/utils/RedirectionApiUtils.java [hunk 3]
```diff
@@ -70,6 +64,60 @@ private NetworkElementEntity makeNetworkElementEntity(NetworkElement networkElem
...
+
+        if (egressEntity == null) {
+            Port egress = this.osClient.networking().port().get(portPair.getIngressId());
+            egressEntity = makeNetworkElementEntity(egress, portPair.getId());
+        }
+
+        return new InspectionPortEntity(portPair.getId(), portPairGroupEntity, ingressEntity, egressEntity);
+    }
+
+    public PortPairGroupEntity makePortPairGroupEntity(PortPairGroup portPairGroup,
```
Comment (arvindn05) [inline nsfc-plugin/src/main/java/org/osc/controller/nsfc/utils/RedirectionApiUtils.java]: mixing removing code with adding code.

File: nsfc-plugin/src/main/java/org/osc/controller/nsfc/api/NeutronSfcSdnRedirectionApi.java [hunk 4]
```diff
@@ -77,62 +69,15 @@ public InspectionPortElement getInspectionPort(InspectionPortElement inspectionP
...
-            }
-        } else {
-            LOG.warn("Failed to retrieve InspectionPort by id! Trying by ingress and egress " + inspectionPort);
-        }
-
-        NetworkElement ingress = inspectionPort.getIngressPort();
-        NetworkElement egress = inspectionPort.getEgressPort();
-
-        return this.utils.findInspectionPortByNetworkElements(ingress, egress);
+        return null; // TODO (Dmitry)
```
Comment (sudhirappaji) [inline nsfc-plugin/src/main/java/org/osc/controller/nsfc/api/NeutronSfcSdnRedirectionApi.java]: Could you give description on TODO, what needs to be done?

**Review by sudhirappaji — APPROVED** [review]
No further comments.

File: nsfc-plugin/src/main/java/org/osc/controller/nsfc/utils/RedirectionApiUtils.java [hunk 5]
```diff
@@ -70,6 +64,17 @@ private NetworkElementEntity makeNetworkElementEntity(NetworkElement networkElem
         return retVal;
     }
 
+    public NetworkElementEntity makeNetworkElementEntity(Port port, String parentId) {
+        throwExceptionIfNullElement(port, "OS Port");
+        List<String> ips;
+        if (port.getFixedIps() != null) {
+            ips = port.getFixedIps().stream().map(ip -> ip.toString()).collect(Collectors.toList());
+        } else {
```
Comment (arvindn05) [inline nsfc-plugin/src/main/java/org/osc/controller/nsfc/utils/RedirectionApiUtils.java]: minor but you can initialize list at 69 and avoid else.

File: nsfc-plugin/src/test/java/org/osc/controller/nsfc/AbstractNeutronSfcPluginTest.java [hunk 6]
```diff
@@ -19,75 +19,65 @@
...
+
     }
 
     @After
     public void tearDown() throws Exception {
-        InMemDB.close();
     }
 
-    protected InspectionHookEntity persistInspectionHook() {
+    protected void persistInspectionHook() {
```
Comment (arvindn05) [inline nsfc-plugin/src/test/java/org/osc/controller/nsfc/AbstractNeutronSfcPluginTest.java]: add a todo here?

persistInspectionHook cannot be same as  persistInspectionPort

File: nsfc-plugin/src/test/java/org/osc/controller/nsfc/OSGiIntegrationTest.java [hunk 7]
```diff
@@ -209,12 +177,7 @@ public SSLContext getSslContext() {
                     mavenBundle("org.apache.servicemix.bundles", "org.apache.servicemix.bundles.dom4j")
                             .versionAsInProject(),
                     mavenBundle("org.javassist", "javassist").versionAsInProject(),
```
Comment (arvindn05) [inline nsfc-plugin/src/test/java/org/osc/controller/nsfc/OSGiIntegrationTest.java:179]: duplicate declaration line 182

**Review by arvindn05 — APPROVED** [review]

File: nsfc-plugin/src/main/java/org/osc/controller/nsfc/api/NeutronSfcSdnControllerApi.java [hunk 8]
```diff
@@ -58,28 +46,11 @@
...
-
-    @Reference(target = "(osgi.jdbc.driver.class=org.h2.Driver)")
-    private DataSourceFactory jdbcFactory;
-
-    @Reference(target = "(osgi.local.enabled=true)")
-    private JPAEntityManagerProviderFactory resourceFactory;
-
     private static final Logger LOG = LoggerFactory.getLogger(NeutronSfcSdnControllerApi.class);
 
     private static final String VERSION = "0.1";
```
Comment (sudhirappaji) [inline nsfc-plugin/src/main/java/org/osc/controller/nsfc/api/NeutronSfcSdnControllerApi.java:51]: I guess this is the version for the plugin. we should enhance the version number, as this is a major enhancement.

Comment (dmitryintel) [reply nsfc-plugin/src/main/java/org/osc/controller/nsfc/api/NeutronSfcSdnControllerApi.java:51]: This one has not been changed for a while. I am not finding any references. It's the same in the NSC plugin

**Pull request merged**